inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+        return "I16";
+    case ::ngraph::element::Type_t::i32:
+        return "I32";
+    case ::ngraph::element::Type_t::i64:
+        return "I64";
+    case ::ngraph::element::Type_t::u8:
+        return "U8";
+    case ::ngraph::element::Type_t::u16:
+        return "U16";
+    case ::ngraph::element::Type_t::u32:
+        return "U32";
+    case ::ngraph::element::Type_t::u64:
+        return "U64";
+    case ::ngraph::element::Type_t::u1:
+        return "BIN";
+    case ::ngraph::element::Type_t::boolean:
+        return "BOOL";
+    default:
+        NGRAPH_CHECK(false, "Incorrect precision: ", elem_type.get_type_name());
+        return "";
+    }
+}
+
+std::string generate_unique_name(
+    const std::unordered_set<std::string>& unique_names, std::string base_name,
+    int suffix) {
+    std::string new_name = base_name + std::to_string(suffix);
+    if (unique_names.find(new_name) == unique_names.end()) {
+        return new_name;
+    } else {
+        suffix++;
+        return generate_unique_name(unique_names, base_name, suffix);
+    }
+}
+
+// TODO: remove when CNNNetwork will be supporting not-unique names
+std::string get_node_unique_name(std::unordered_set<std::string>& unique_names,
+                                 const ngraph::Node* n) {
+    std::string name = n->get_friendly_name();
+    if (unique_names.find(name) != unique_names.end()) {
+        name = generate_unique_name(unique_names, name, 0);
+    }
+    unique_names.insert(name);
+    return name;
+}
+
+void ngfunction_2_irv10(pugi::xml_document& doc, std::vector<uint8_t>& bin,
+                        const ngraph::Function& f) {
+    pugi::xml_node netXml = doc.append_child("net");
+    netXml.append_attribute("name").set_value(f.get_friendly_name().c_str());
+    netXml.append_attribute("version").set_value("10");
+    pugi::xml_node layers = netXml.append_child("layers");
+
+    const std::map<ngraph::Node*, int> layer_ids = create_layer_ids(f);
+    std::unordered_set<std::string> unique_names;
+
+    for (auto n : f.get_ordered_ops()) {
+        ngraph::Node* node = n.get();
+
+        NGRAPH_CHECK(layer_ids.find(node) != layer_ids.end(),
+                     "Internal error.");
+        // <layers>
+        pugi::xml_node layer = layers.append_child("layer");
+        layer.append_attribute("id").set_value(layer_ids.find(node)->second);
+        layer.append_attribute("name").set_value(
+            get_node_unique_name(unique_names, node).c_str());
+        layer.append_attribute("type").set_value(get_type_name(node).c_str());
+        layer.append_attribute("version").set_value(
+            get_opset_name(node).c_str());
+
+        // <layers/data>
+        pugi::xml_node data = layer.append_child("data");
+
+        // <layers/data> general atributes
+        XmlVisitor visitor{data};
+        NGRAPH_CHECK(node->visit_attributes(visitor),
+                     "Visitor API is not supported in ", node->get_name(),
```

Comment:
`get_name` will return some internal ngraph name that you won't be able to map to original node.
Let's use operator<< for node instead.

Reply:
this is applicable for all NGRAPH_CHECK places

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/48104070eae5a01692536a45e7591760ac7e06d6)